var/spack/repos/builtin/packages/py-onnx-runtime/package.py

```diff
@@ -106,16 +109,13 @@ def cmake_args(self):
 
         return args
 
-    def setup_file(self):
-        return join_path(self.stage.source_path, 'setup.py')
-
-    @run_after('build')
-    def build_python(self):
-        """Build everything needed to install."""
-        with working_dir(self.stage.source_path):
-            PythonPackage.build(self, self.spec, self.prefix)
-
     @run_after('install')
     def install_python(self):
-        with working_dir(self.stage.source_path):
-            PythonPackage.install(self, self.spec, self.prefix)
+        # Notice: this is a very ugly hack, but for some reason
+        # wrapping pip invocation in `with working_dir` does
+        # not work - setup.py can't find onnxruntime/backend directory
+        args = ['-c', 'cd', self.build_directory, '&&', pip.command]
+        args.extend(PythonPackage._std_args(self))
+        args.append('--prefix=' + self.spec.prefix)
+        bash = which('bash')
+        bash(*args)
```

Comment:
Does this work?
```suggestion
        args = std_pip_args + ['--prefix=' + prefix, '.']
        pip(*args)
```
This is the same thing the base class runs but without any change of directory. Using:
```python
with working_dir(...):
    PythonPackage.install(...)
```
doesn't work because `install` also has its own `with working_dir`.

Reply:
Nope, doesn't work as well:
```
==> [2022-03-23-11:53:58.172250] '/home/razumov/my_spack/opt/spack/linux-fedora35-skylake/gcc-11.2.1/python-3.9.10-7zu5c3wpgyypwjwdq7qx42ib4ee5lzmk/bin/python3.9' '-m' 'pip' '-vvv' '--no-input' '--no-cache-dir' '--disable-pip-version-check' 'install' '--no-deps' '--ignore-installed' '--no-build-isolation' '--no-warn-script-location' '--no-index' '--prefix=/home/razumov/my_spack/opt/spack/linux-fedora35-skylake/gcc-11.2.1/py-onnx-runtime-1.7.2-jcp7afthhelar3qddcyr7ns4n3cvlbvq' '.'
Using pip 21.3.1 from /home/razumov/my_spack/opt/spack/linux-fedora35-skylake/gcc-11.2.1/py-pip-21.3.1-jsxaxtghz575vhuhh2vpowx2zo44vzed/lib/python3.9/site-packages/pip (python 3.9)
Non-user install due to --prefix or --target option
Ignoring indexes: https://pypi.org/simple
Created temporary directory: /tmp/pip-ephem-wheel-cache-pi2_p5g6
Created temporary directory: /tmp/pip-req-tracker-ftwrzmi_
Initialized build tracking at /tmp/pip-req-tracker-ftwrzmi_
Created build tracker: /tmp/pip-req-tracker-ftwrzmi_
Entered build tracker: /tmp/pip-req-tracker-ftwrzmi_
Created temporary directory: /tmp/pip-install-_fnk6uyn
Processing /tmp/razumov/spack-stage/spack-stage-py-onnx-runtime-1.7.2-jcp7afthhelar3qddcyr7ns4n3cvlbvq/spack-src
  Added file:///tmp/razumov/spack-stage/spack-stage-py-onnx-runtime-1.7.2-jcp7afthhelar3qddcyr7ns4n3cvlbvq/spack-src to build tracker '/tmp/pip-req-tracker-ftwrzmi_'
  Running setup.py (path:/tmp/razumov/spack-stage/spack-stage-py-onnx-runtime-1.7.2-jcp7afthhelar3qddcyr7ns4n3cvlbvq/spack-src/setup.py) egg_info for package from file:///tmp/razumov/spack-stage/spack-stage-py-onnx-runtime-1.7.2-jcp7afthhelar3qddcyr7ns4n3cvlbvq/spack-src
  Created temporary directory: /tmp/pip-pip-egg-info-sx3k5phr
  Preparing metadata (setup.py): started
  Running command python setup.py egg_info
  running egg_info
  creating /tmp/pip-pip-egg-info-sx3k5phr/onnxruntime.egg-info
  writing /tmp/pip-pip-egg-info-sx3k5phr/onnxruntime.egg-info/PKG-INFO
  writing dependency_links to /tmp/pip-pip-egg-info-sx3k5phr/onnxruntime.egg-info/dependency_links.txt
  writing entry points to /tmp/pip-pip-egg-info-sx3k5phr/onnxruntime.egg-info/entry_points.txt
  writing requirements to /tmp/pip-pip-egg-info-sx3k5phr/onnxruntime.egg-info/requires.txt
  writing top-level names to /tmp/pip-pip-egg-info-sx3k5phr/onnxruntime.egg-info/top_level.txt
  writing manifest file '/tmp/pip-pip-egg-info-sx3k5phr/onnxruntime.egg-info/SOURCES.txt'
  error: package directory 'onnxruntime/backend' does not exist
  Preparing metadata (setup.py): finished with status 'error'
```

Reply:
Okay, what about `PythonPackage.install(self, self.spec, self.prefix)` but also add `build_directory = '.'` to the package-level settings? Then we build in the source dir instead of in a custom build dir.

Reply:
Yes, this way it works. Thanks!